addons/binding/org.openhab.binding.konnected/README.md

```diff
@@ -0,0 +1,65 @@
+Konnected is an open-source firmware and software that runs on a NodeMCU ESP8266 device. 
+The Konnected hardware is specifically designed for an alarm panel installation, but the general purpose firmware/software can be run on any ESP8266 device.
+https://Konnected.io/_
+
+## Supported Things
+
+This binding supports one type of thing:  a Konnected module.
+
+## Discovery
+
+The binding will auto discover Konnected modules which are attached to the same network as the server running openHAB via UPnP.  
+The binding will then create things for each module discovered which can be added.
+
+## Binding Configuration
+
+There is no configuration required for the binding.
+
+## Thing Configuration
+
+The auto-discovery service of the binding will detect the ipAddress and port of the Konnected module.  
+But once it is added you will need to provide an Authority Token to secure communication between the module and openHAB.  
+The binding will attempt to discover the ip address of your OpenHAB server.  However, if it is unable to determine the ip address you can manually define the ip address and port in the thing configuration.
+
+## Channels
+
+The auto discovered thing adds two default channels.
+One channel for Zone 6 which is a sensor type channel, and one channel for the out pin that is an actuator type channel.
+These channels represent the two pins on the Konnected module whose type cannot be changed.
+For zones 1-5, you will need to add channels for the remaining zones that you have connected.
+The only requirement is that the last character of the typeId is the corresponding zone number in integer format.  
+For example, a sensor hooked up to zone 1.  A new channel of type sensor with a typeID of any of the following names would work:
+Zone1
+Zone_1
+Sensor1
+Pin1
+Pin_1
+1
+
+Then you need to link the corresponding item to the channel.
+For sensor type channels the channel type is the contact item.
+For the actuator type channel, the channel type is switch item. 
+For the actuator type channels you can also add configuration parameters times, pause and momentary which will be added to the payload that is sent to the Konnected Module.
+These parameters will tell the module to pulse the actuator for certain time period.
+A momentary switch actuates a switch for a specified time (in milliseconds) and then reverts it back to the off state. This is commonly used with a relay module to actuate a garage door opener, or with a doorbell to send a momentary trigger to sound the doorbell.
+A beep/blink switch is like a momentary switch that repeats either a specified number of times or indefinitely. This is commonly used with a a piezo buzzer to make a "beep beep" sound when a door is opened, or to make a repeating beep pattern for an alarm or audible warning. It can also be used to blink lights.
+
+## Full Example
+
+_Provide a full usage example based on textual configuration files (*.things, *.items, *.sitemap)._
```

Comment:
So maybe i need to change some other aspects of the binding, or better document this in the readme.  But all  thing discovery as it stands right now is handled in the binding via the upnp service.  When i was trying to see how you would model the things file i don't see where i would specify the properties in the things file only the configuration parameters.  Do i need to code the binding to match the properties? 

https://www.openhab.org/docs/configuration/things.html#defining-things

That is why I did not have a things file in the readme, as all thing creation is being handled by the binding itself.  The modules are found via the upnp service which then has the appropriate ip address property where they can be reached.

Reply:
Ok I understand now.  So each thing does have its own callback URL.  The urls are presently formed by the binding with the base formula of:

http://((address_of_service_obtained_from_service) or (config.ipaddress of thing if defined))/Konnected/thing_id

So what you are suggesting is a configuration parameter that could replace /Konnected/thing_id and if none is provided in the thing then use the defaults?  

I had simply hard_coded in the path of /Konnected and then taken the /id part from the module name which is also "hard coded" so your suggestion is simply to give the user an option to configure these and otherwise default to what i was using.  Makes sense.

Reply:
The only property of a thing is its ipAddress and serial number (macAddress) obtained via the uPnP service.  

Which as a side note i am not really sure if i am handling the situation where the ipAddress of a module changes with the UPNP service. The best practices of for the konnected modules has you set them to a static ip address so its not a situation that i have run into during development yet but it one that i would want to deal with where the ipAddress would change but the thing would need to stay the same since the user would have already configured all of their channels.